packages/deployments/contracts/contracts/interpreters/Executor.sol

```diff
@@ -11,92 +13,138 @@ import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
-   *        calldata execution
-   * @param to The address to execute the calldata on
-   * @param assetId The assetId of the funds to approve to the contract or
-   *                send along with the call
-   * @param amount The amount to approve or send with the call
-   * @param callData The data to execute
+   * call data executes can be payable, and will have `amount` sent
+   * along with the function (or approved to the contract). If the
+   * call fails, rather than reverting, funds are sent directly to 
+   * some provided fallbaack address
+   * @param _transferId Unique identifier of transaction id that necessitated
+   * calldata execution
+   * @param _amount The amount to approve or send with the call
+   * @param _to The address to execute the calldata on
+   * @param _assetId The assetId of the funds to approve to the contract or
+   * send along with the call
+   * @param _properties The origin properties
+   * @param _callData The data to execute
    */
   function execute(
-    bytes32 transferId,
-    address payable to,
-    address assetId,
-    uint256 amount,
-    bytes calldata callData
-  ) override external payable onlyConnext returns (bool, bool, bytes memory) {
+    bytes32 _transferId,
+    uint256 _amount,
+    address payable _to,
+    address _assetId,
+    bytes memory _properties,
+    bytes calldata _callData
+  ) override external payable onlyConnext returns (bool, bytes memory) {
     // If it is not ether, approve the callTo
     // We approve here rather than transfer since many external contracts
     // simply require an approval, and it is unclear if they can handle 
     // funds transferred directly to them (i.e. Uniswap)
-    bool isNative = assetId == address(0);
+    bool isNative = _assetId == address(0);
     if (!isNative) {
-      SafeERC20Upgradeable.safeIncreaseAllowance(IERC20Upgradeable(assetId), to, amount);
+      SafeERC20Upgradeable.safeIncreaseAllowance(IERC20Upgradeable(_assetId), _to, _amount);
     }
 
     // Check if the callTo is a contract
     bool success;
     bytes memory returnData;
-    bool isContract = AddressUpgradeable.isContract(to);
-    if (isContract) {
-      // Try to execute the callData
-      // the low level call will return `false` if its execution reverts
-      (success, returnData) = to.call{value: isNative ? amount : 0}(callData);
-    }
+    require(AddressUpgradeable.isContract(_to), "!contract");
+    
+    // If it should set the properties, set them.
+    // NOTE: safe to set the properties always because modifier will revert if
+    // it is the wrong type on conversion, and revert occurs with empty type as
+    // well
+    properties = _properties;
+  
+    // Try to execute the callData
+    // the low level call will return `false` if its execution reverts
+    (success, returnData) = _to.call{value: isNative ? _amount : 0}(_callData);
+    
+    // Unset properties
+    properties = LibCrossDomainProperty.EMPTY_BYTES;
```

Comment:
this usage of temporary storage `properties` is cool... I can't help but wonder if it would be more gas efficient to optionally allow the user to specify a type of an interpreter layer that accepts an additional argument, like msg.sender, but I think there's additional complexity that comes with that.

Reply:
This is so you can allow the `to` contract to do this:

```ts
IExecutor(msg.sender).originSender()
IExecutor(msg.sender).origin()
```

within whatever calldata you're passing through. adding in params makes it difficult because you wouldn't be able to use the low-level `call` (can't insert params there) so you would be forcing everyone to write a middleware contract

Reply:
that being said, it likely would be more gas efficient in cases where you don't need to check those properties

Reply:
yeah, makes sense! will properties *always* include `msg.sender` then? are there any other properties that will for sure be specified? lol